include/ginkgo/core/matrix/fbcsr.hpp

```diff
@@ -0,0 +1,390 @@
+     */
+    void sort_by_column_index();
+
+    /*
+     * Tests if all row entry pairs (value, col_idx) are sorted by column index
+     *
+     * @returns True if all row entry pairs (value, col_idx) are sorted by
+     *          column index
+     */
+    bool is_sorted_by_column_index() const;
+
+    /**
+     * @return The values of the matrix.
+     */
+    value_type *get_values() noexcept { return values_.get_data(); }
+
+    /**
+     * @see Fbcsr::get_const_values()
+     */
+    const value_type *get_values() const noexcept
+    {
+        return values_.get_const_data();
+    }
+
+    /**
+     * @copydoc Fbcsr::get_values()
+     *
+     * @note This is the constant version of the function, which can be
+     *       significantly more memory efficient than the non-constant version,
+     *       so always prefer this version.
+     */
+    const value_type *get_const_values() const noexcept
+    {
+        return values_.get_const_data();
+    }
+
+    /**
+     * @return The column indexes of the matrix.
+     */
+    index_type *get_col_idxs() noexcept { return col_idxs_.get_data(); }
+
+    /**
+     * @see Fbcsr::get_const_col_idxs()
+     */
+    const index_type *get_col_idxs() const noexcept
+    {
+        return col_idxs_.get_const_data();
+    }
+
+    /**
+     * @copydoc Fbcsr::get_col_idxs()
+     *
+     * @note This is the constant version of the function, which can be
+     *       significantly more memory efficient than the non-constant version,
+     *       so always prefer this version.
+     */
+    const index_type *get_const_col_idxs() const noexcept
+    {
+        return col_idxs_.get_const_data();
+    }
+
+    /**
+     * @return The row pointers of the matrix.
+     */
+    index_type *get_row_ptrs() noexcept { return row_ptrs_.get_data(); }
+
+    /**
+     * @see Fbcsr::get_const_row_ptrs()
+     */
+    const index_type *get_row_ptrs() const noexcept
+    {
+        return row_ptrs_.get_const_data();
+    }
+
+    /**
+     * @copydoc Fbcsr::get_row_ptrs()
+     *
+     * @note This is the constant version of the function, which can be
+     *       significantly more memory efficient than the non-constant version,
+     *       so always prefer this version.
+     */
+    const index_type *get_const_row_ptrs() const noexcept
+    {
+        return row_ptrs_.get_const_data();
+    }
+
+    /**
+     * Returns the number of elements explicitly stored in the matrix.
+     *
+     * @return the number of elements explicitly stored in the matrix
+     */
+    size_type get_num_stored_elements() const noexcept
+    {
+        return values_.get_num_elems();
+    }
+
+    /**
+     * @return The fixed block size for this matrix
+     */
+    int get_block_size() const { return bs_; }
+
+    /**
+     * Set the fixed block size for this matrix
+     *
+     * @param block_size The block size
+     */
+    void set_block_size(const int block_size) { bs_ = block_size; }
+
+    /**
```

Comment:
Is it a group policy to always have a "description" even if clearly unnecessary? Why? Some of the other functions in this file only have them because they were copied from `Csr` by the script. :sweat_smile:

Reply:
IMO those lines are unnecessary even in the final documentation. It's literally the same sentence repeated twice.